src/OSPSuite.R/Services/DataImporter.cs

```diff
@@ -0,0 +1,198 @@
+
+         var compCategory = createMetaDataCategory<string>(Constants.ObservedData.COMPARTMENT, isMandatory: true, isListOfValuesFixed: true);
+         compCategory.Description = ObservedData.ObservedDataCompartmentDescription;
+         compCategory.TopNames.Add(Constants.ObservedData.PLASMA_COMPARTMENT);
+         categories.Add(compCategory);
+
+         var moleculeCategory = createMetaDataCategory<string>(Constants.ObservedData.MOLECULE);
+         moleculeCategory.Description = ObservedData.MoleculeNameDescription;
+         moleculeCategory.AllowsManualInput = true;
+         categories.Add(moleculeCategory);
+
+         // Add non-mandatory metadata categories
+         var molecularWeightCategory = createMetaDataCategory<double>(Constants.ObservedData.MOLECULARWEIGHT);
+         molecularWeightCategory.MinValue = 0;
+         molecularWeightCategory.MinValueAllowed = false;
+         categories.Add(molecularWeightCategory);
+         categories.Add(createMetaDataCategory<string>(Constants.ObservedData.STUDY_ID));
+         categories.Add(createMetaDataCategory<string>(Constants.ObservedData.SUBJECT_ID));
+         categories.Add(createMetaDataCategory<string>(Constants.ObservedData.GENDER, isListOfValuesFixed: true));
+         categories.Add(createMetaDataCategory<string>(Constants.ObservedData.DOSE));
+         categories.Add(createMetaDataCategory<string>(Constants.ObservedData.ROUTE));
+         categories.Add(createMetaDataCategory<string>(Constants.ObservedData.GROUP_ID));
+
+         return categories;
+      }
+
+      public (IReadOnlyList<DataRepository> DataRepositories, ImporterConfiguration Configuration) ImportDataSets(
+         IReadOnlyList<MetaDataCategory> metaDataCategories, 
+         IReadOnlyList<ColumnInfo> columnInfos, 
+         DataImporterSettings dataImporterSettings,
+         string dataFileName)
+      {
+         throw new NotImplementedException();
+      }
+
+      public IReadOnlyList<DataRepository> ImportFromConfiguration(
+         ImporterConfiguration configuration, 
+         IReadOnlyList<MetaDataCategory> metaDataCategories, 
+         IReadOnlyList<ColumnInfo> columnInfos, 
+         DataImporterSettings dataImporterSettings,
+         string dataFileName)
+      {
+         if (string.IsNullOrEmpty(dataFileName))
+            return Enumerable.Empty<DataRepository>().ToList();
+
+         try
+         {
+            var importedData = _importer.ImportFromConfiguration(configuration, columnInfos, dataFileName, metaDataCategories, dataImporterSettings);
+            if (importedData.MissingSheets.Count != 0)
+               _dialogCreator.MessageBoxError(Captions.Importer.SheetsNotFound(importedData.MissingSheets));
+            return importedData.DataRepositories.Select(drm => drm.DataRepository).ToList();
+         }
+         catch (Exception e) when (e is UnsupportedFormatException || e is UnsupportedFileTypeException)
+         {
+            _dialogCreator.MessageBoxError(e.Message);
```

Comment:
same

Reply:
Here, we do need to show a feedback to the user somehow. Is your suggestion to let it break and show the exception from the R side? In general I could just leave the exceptions unhandled and that would do the work, right?

Reply:
>In general I could just leave the exceptions unhandled and that would do the work, right?

yes

Reply:
In general, _dialogCreator is not necessarily showing a message box. It could simply be showing the error to the console if there is an implementation for this. In any case, throwing exception might NOT be what we want here depending on how it's handled. We do not want to see a stack trace in R so maybe using the logger would be better (catch, log, return null or empty or whatever )